core/src/test/scala/akka/persistence/cassandra/EventsByTagRestartSpec.scala

```diff
@@ -80,5 +84,87 @@ class EventsByTagRestartSpec extends CassandraSpec(EventsByTagRestartSpec.config
       tagProbe.expectNoMessage(waitTime)
       tagProbe.cancel()
     }
+
+    "not write incorrect tag sequence nrs when actor is stopped prematurely" in {
+      val tag = "green"
+      val queryJournal = PersistenceQuery(system).readJournalFor[CassandraReadJournal](CassandraReadJournal.Identifier)
+      val p2 = system.actorOf(TestTaggingActor.props("p2", Set(tag)))
+      val probe1 = TestProbe()
+      probe1.watch(p2)
+      p2.tell("e1", probe1.ref)
+      probe1.expectMsg(Ack)
+      p2.tell("e2", probe1.ref)
+      probe1.expectMsg(Ack)
+      p2.tell("e3", probe1.ref)
+      probe1.expectMsg(Ack)
+      p2.tell("e4", probe1.ref)
+
+      // the PoisonPill will cause the actor to be stopped before the journal write has been completed,
+      // and is therefore similar to if the CircuitBreaker would trigger
+      p2 ! PoisonPill
+      probe1.expectTerminated(p2)
```

Comment:
It's possible that the write is completed before poison/termination and then the probe will receive an Ack here instead of Terminated. Maybe have to use two different probes to ignore the Ack?

Same in next test case.

```
[info] - must use correct tag sequence nrs when actor is stopped prematurely and then restarted *** FAILED *** (52 milliseconds)
1340[info]   java.lang.AssertionError: assertion failed: expected: Terminated Actor[akka://EventsByTagRestartSpec/user/$l#-1513242516] but got unexpected message Ack
1341[info]   at scala.Predef$.assert(Predef.scala:282)
1342[info]   at akka.testkit.TestKitBase.expectMsgPF(TestKit.scala:437)
1343[info]   at akka.testkit.TestKitBase.expectMsgPF$(TestKit.scala:433)
1344[info]   at akka.testkit.TestKit.expectMsgPF(TestKit.scala:928)
1345[info]   at akka.testkit.TestKitBase.expectTerminated(TestKit.scala:451)
1346[info]   at akka.testkit.TestKitBase.expectTerminated$(TestKit.scala:450)
1347[info]   at akka.testkit.TestKit.expectTerminated(TestKit.scala:928)
1348[info]   at akka.persistence.cassandra.EventsByTagRestartSpec.$anonfun$new$9(EventsByTagRestartSpec.scala:145)
```